crates/factor-outbound-http/src/wasi.rs

```diff
@@ -127,12 +148,6 @@ async fn send_request_impl(
 
         let path_and_query = request.uri().path_and_query().cloned();
         *request.uri_mut() = origin.into_uri(path_and_query);
-    } else {
-        let outbound_url = OutboundUrl::parse(request.uri().to_string(), "https")
-            .map_err(|_| ErrorCode::HttpRequestUriInvalid)?;
-        if !allowed_hosts.allows(&outbound_url) {
-            return Ok(handle_not_allowed(request.uri(), false));
-        }
     }
 
     if let Some(authority) = request.uri().authority() {
```

Comment:
Nit: I know this was here before, but it seems strange we have two different blocks for when `request.uri().authority()` is `Some` instead of just combining them.

Reply:
Hmm...authority _should_ always be set here. I'll change it to an error.